File: dec7/two.mjs
import fs from "fs/promises";

(async () => {
    const file = (await fs.readFile("input.txt", "utf-8")).trim();
    let sum = 0;    
    file.split("\n").forEach(line => {
        let [res, nums] = line.split(": ");
        res = Number(res);
        nums = nums.split(" ").map(Number);
        for (let i = 0, len = nums.length - 1; i < 3 ** len; i++) {
            let value = nums[0];
            let currentI = i;
            for (let j = 0; j < len; j++) {
                const op = currentI % 3;
                currentI = (currentI / 3) | 0;
                if (op === 0) {
                    value += nums[j + 1];
                } else if (op === 1) {
                    value *= nums[j + 1];
                } else {
                    value = +(String(value) + String(nums[j + 1]));
                }
            }
            if (value === res) {
                sum += res;
                break;
            }
        }
    });
    console.log(sum);
})();

File: dec7/two.rs
use std::fs;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let file = fs::read_to_string("input.txt")?.trim().to_string();
    let mut sum = 0;
    for line in file.lines() {
        let (res_str, nums_str) = line.split_once(": ").unwrap();
        let res: i64 = res_str.parse().unwrap();
        let nums: Vec<i64> = nums_str.split_whitespace().map(|s| s.parse().unwrap()).collect();
        let len = nums.len() - 1;
        for i in 0..3_i64.pow(len as u32) {
            let mut value = nums[0];
            let mut current_i = i;
            for j in 0..len {
                let op = current_i % 3;
                current_i /= 3;
                match op {
                    0 => value += nums[j + 1],
                    1 => value *= nums[j + 1],
                    _ => {
                        let value_str = format!("{}{}", value, nums[j + 1]);
                        value = value_str.parse().unwrap();
                    }
                }
            }
            if value == res {
                sum += res;
                break;
            }
        }
    }
    println!("{}", sum);
    Ok(())
}